graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -61,6 +65,55 @@ def _validate_macro_ast_with_macro_directives(schema, ast, macro_directives):
                 .format(directive_definition.name, len(macro_data)))
 
 
+def _validate_macro_ast_directives(ast, inside_fold_scope=False):
+    """Check that the macro is using non-macro direcives properly.
+
+    Restrictions on use of directives:
+    - @output and @output_source are disallowed
+    - @macro_edge_target is not allowed to be inside a @fold scope
+
+    Args:
+        ast: GraphQL AST describing a subtree of the macro
+        inside_fold_scope: bool, whether the subtree is within a @fold scope
+    """
+    subselection_inside_fold_scope = inside_fold_scope
+    directives_with_no_restrictions = (
```

Comment:
nit: We're using these with the `in` operator, any reason they shouldn't be frozensets instead?